snapcraft/tests/fake_servers.py

```diff
@@ -426,6 +430,59 @@ def _handle_sign_build_request(self):
         self.end_headers()
         self.wfile.write(content)
 
+    def _handle_close_request(self):
+        logger.debug('Handling close request')
+        string_data = self.rfile.read(
+            int(self.headers['Content-Length'])).decode('utf8')
+        payload = json.loads(string_data)
+        channels = payload['channels']
+
+        if channels == ['invalid']:
+            self.send_response(400)
+            self.send_header('Content-Type', 'application/json')
+            error = {
+                'error_list': [
+                    {'code': 'invalid-field',
+                     'message': ('The \'channels\' field content is not '
```

Comment:
Not related to the mock, but would be nice if the store returned a list of valid channels to use.